pandas/_libs/tslibs/timedeltas.pyx

```diff
@@ -1520,6 +1529,10 @@ class Timedelta(_Timedelta):
         div = other // self
         return div, other - div * self
 
+    # GH40946
+    def total_seconds(self):
+        return self.value / 1e9
```

Comment:
internally, ns precision is achieved by storing the value as a 64 bit integer.

once this is converted to a float, we cannot maintain ns precision. However, including the nanosecond component is probably more accurate than ignoring it.

I think we need more discussion on the issue before re-instating the nanosecond "precision". 

from https://github.com/pandas-dev/pandas/pull/31380#discussion_r374120257

> ... from "dubiously nanosecond-precision" to "microsecond-precision" ...

The change to ms precision was an intentional one but was not documented.

Changing it back would probably require a specific mention in the release note.

Also the tests are failing for the cases added when the precision of `total_seconds` changed, so this needs to be fixed also.

I also think that the 2 issue should be addressed independently.

Reply:
Before I go and refactor/enhance the tests, it would be nice to have a categorical decision on this topic. Should the `total_seconds` part be a separate PR? I am not really sure why the nanosecond precision is a problem. The precise epoch in ns can be obtained by the `value` property.

Reply:
> Should the `total_seconds` part be a separate PR?

the two fixes are orthogonal? The override of the `total_seconds` method to fix #40946 and the change to `__new__` to fix the constructor issue https://github.com/pandas-dev/pandas/issues/43764#issuecomment-927709183?

If so, I think should be two separate PRs.

Reply:
yes pls split to a sepearate PR